File: src/main/ruby/com_github_leetcode/list_node.rb
# Definition for singly-linked list.
class ListNode
  attr_accessor :val, :next

  def initialize(x = 0, _next = nil)
    @val = x
    @next = _next
  end

  def to_a
    result = []
    current = self
    while current
      result << current.val
      current = current.next
    end
    result
  end
end

File: src/main/js/com_github_leetcode/listnode.js
class ListNode {
    constructor(val, next) {
        this.val = val === undefined ? 0 : val
        this.next = next === undefined ? null : next
    }

    toString() {
        let result = `${this.val}`
        let current = this.next
        while (current !== null) {
            result += `, ${current.val}`
            current = current.next
        }
        return result
    }
}

export { ListNode }
